### ISSUE TITLE
Sass support

### ISSUE THREAD

**Issue by TheJaredWilcurt** [filearts/plunker_run#7]
Codepen and JSBin both suppport SCSS and Sass syntax for CSS pre-processing. JSFiddle supports just SCSS. Sassmeister is open source and supports pretty much everything related to Sass, so they may be a good place to steal code from.

* https://github.com/jedfoster/SassMeister

**Comment by ggoodman** [comment]
Hi @TheJaredWilcurt, Plunker has supported SASS and SCSS for quite some time. Less as well.

Create a file named `style.scss`, but request `style.css` in your html. Plunker will do the transformation for you on the run server. Please let me know if this works as expected as I recently rewrote the run server (see https://github.com/ggoodman/plunker-run-plugin).

**Comment by tkrotoff** [comment]
@ggoodman it does not work: http://plnkr.co/edit/FmNtus?p=preview

**Comment by ggoodman** [comment]
Fixed by ggoodman/plunker-run-plugin@1451921843a8cc80e426037cd723a693039be8da

**Comment by TheJaredWilcurt** [comment]
@ggoodman 

Still doesn't work. Also sass files lack syntax highlighting:
- http://plnkr.co/edit/07c5RgqcEAx9in7FZUhN?p=preview

**Comment by tkrotoff** [comment]
@TheJaredWilcurt your Plunker is incorrect.
You didn't get how it works: when Plunker sees a file ending with .scss it compiles it to .css. So in your HTML you should only reference the generated .css file, not the .scss file.

In your Plunker:

``` HTML
<link rel="stylesheet" href="style.css" />
<link rel="stylesheet" href="style.scss" />
<link rel="stylesheet" href="style.sass" />
```

should be written:

```
<link rel="stylesheet" href="style.css">
```

And you should have only 1 file style.\* (style.scss or style.css). Here you have 2 files style.\* (style.sass and style.scss).
With these fixes your Plunker works fine: http://plnkr.co/edit/OEuhs7rnCRRhTHtHLgOM?p=preview

btw, stop writing `/>` in HTML5 => not needed + ugly :)

Edit: my previous Plunker now works: http://plnkr.co/edit/FmNtus?p=preview so the issue has been fixed.

**Comment by ggoodman** [comment]
This should clear up how to use compilation in Plunker: http://embed.plnkr.co/suu7Yg/preview

**Comment by TheJaredWilcurt** [comment]
@tkrotoff

...yeah.. I **\* get that. I've tried everything. The point of including all 3 was to show that literally nothing works. And I still don't see this working.

I see that you've some how found a way to get it to work, however, when I straight up copy and paste your code, it doesn't.

btw, *************, all tags should be closed, like your ugly mouth.

---

@ggoodman 

> "Plunker is designed to make it as easy as possible for you to work in compiled web languages."

This is not easy. This is not intuitive. This is actually really frustrating. I never had to spend 20 minutes failing on Codepen or JSBin or JSFiddle. All of them support Sass with no effort at all on the users part. Why is this so terrible? I STILL can't get it to work and I'm about 100% sure you guys don't actually support .sass files. There's no syntax highlighting and when I rename the .scss file to .sass and correct the syntax in the [linked project](http://plnkr.co/edit/FmNtus?p=preview), it doesn't work.

If you want to properly simulate a real project that uses sass, then you need to support folders in plunkr. There should be a sass folder and the contents of it should be processed and spit out in a style or css folder that is then linked to by the html.

Creating a page that no one will ever see explaining how your site is SUPPOSED to work and then wiping off your hands and saying "Well this should fix everything, they clearly just don't know how I want it to work"... isn't a solution. Maybe you should actually FIX the site so it does what you're telling people it does. Better yet, design it better so no one has to be "taught" how to use it. Sites should be intuitive and if they aren't, people don't use them.

> "A life without Sass, is a life not worth coding" - Abraham Lincoln, inventor of ultra freedom

If there really is a way to do this, and it actually isn't broken, then make some fucking templates so people don't have to go through this ridiculous unintuitive process. If there was just a "Sass" and a "SCSS" template available I would have just used those and edited them for my needs as they would work out of the box.

---

And then you make some remark about "blah blah blah, you're being rude"

Ya know what's ****\* rude. I point out a feature without which your site is pointless to me, and am lied to and told it works and have the issue instantly closed. Then that is repeated again, and again.

**Comment by tkrotoff** [comment]
@TheJaredWilcurt 

> fuck off [...] ugly mouth

I give you a lot of love and kisses too.

> I never had to spend 20 minutes failing on Codepen or JSBin or JSFiddle

Because they have a different approach. They provide 1 HTML "file", 1 CSS "file", 1 JS "file" whereas with Plunker you can create as many files as you want like you would do in real life.
And in real life you have to preprocess .scss files to generate .css files and reference them in your `<head>`.

Due to the "1 HTML file, 1 CSS file, 1 JS file" approach, Codepen/JSBin/JSFiddle already handle this for you (you don't write the `<head>` part, all this complexity is hidden from you).

To sum up:
- Plunker => more powerful but more responsibility for the developer
- Codepen/JSBin/JSFiddle => less powerful but easier

What can be improved:
- Support .sass (only .scss is currently supported), meanwhile display a warning if .sass
- Display a warning when someone write `<link rel="stylesheet" href="style.scss">` instead of `<link rel="stylesheet" href="style.css">`

PS: I'm not affiliated with Plunker, just a random user.

Edit:

> [...] and told it works and have the issue instantly closed

Instantly in the first place then re-opened => a commit has been made to fix the .scss support.

**Comment by ggoodman** [comment]
I'm sorry to have invested my free time in creating a service that wasted a whole 20m of your time! Next time I will be more judicious in the use of my time. There is no team here, this was built by a self-taught programmer in his spare time who, at the time, had no professional programming experience.

I've done my best to provide a user-friendly and intuitive way to navigate the complexity of compiled languages in an environment in which I have much less control over how and what gets rendered. If it is not meeting your needs, perhaps future work will make the support better.

As to documentation, it is a known and regrettable short-coming of the website. 

> Support .sass (only .scss is currently supported), meanwhile display a warning if .sass

I was not aware that `.sass` syntax was broken. Perhaps someone could take a look at the source and propose a fix:
https://github.com/ggoodman/plunker-run-plugin/blob/master/adapters/transformers/sass.js

> Display a warning when someone write <link rel="stylesheet" href="style.scss"> instead of <link rel="stylesheet" href="style.css">

I prefer not to make any assumptions about what users' intentions are. Catching every corner-case like this is well beyond the maintenance and time requirements I'm able to invest.

Thank you @tkrotoff for the thoughtful and polite response; you have understood some of the trade-offs that come with a more bare-metal approach to online code editors.

**Comment by tkrotoff** [comment]
> more bare-metal approach to online code editors

And I love this approach! I can easily copy-paste and re-create the structure of real projects.

![](http://www.cinemablend.com/images/sections/67848/care_bears_67848.jpg)
